core/orchestrator.go

```diff
@@ -485,12 +482,86 @@ func (rt *RemoteTranscoder) Transcode(fname string, profiles []ffmpeg.VideoProfi
 	return [][]byte{}, fmt.Errorf("Unknown error")
 }
 
-func NewRemoteTranscoder(n *LivepeerNode, stream net.Transcoder_RegisterTranscoderServer) *RemoteTranscoder {
+func NewRemoteTranscoder(n *LivepeerNode, stream net.Transcoder_RegisterTranscoderServer, capacity int) *RemoteTranscoder {
 	return &RemoteTranscoder{
-		node:   n,
-		stream: stream,
-		eof:    make(chan struct{}, 1),
+		node:     n,
+		stream:   stream,
+		eof:      make(chan struct{}, 1),
+		capacity: capacity,
+	}
+}
+
+func NewRemoteTranscoderManager() RemoteTranscoderManager {
+	return RemoteTranscoderManager{
+		remoteTranscoders: []*RemoteTranscoder{},
+		liveTranscoders:   map[net.Transcoder_RegisterTranscoderServer]*RemoteTranscoder{},
+		RTmutex:           &sync.Mutex{},
+	}
+}
+
+type RemoteTranscoderManager struct {
+	remoteTranscoders []*RemoteTranscoder
+	liveTranscoders   map[net.Transcoder_RegisterTranscoderServer]*RemoteTranscoder
+	RTmutex           *sync.Mutex
+}
+
+func (rtm *RemoteTranscoderManager) Register(transcoder *RemoteTranscoder) {
+	rtm.RTmutex.Lock()
+	defer rtm.RTmutex.Unlock()
+	rtm.liveTranscoders[transcoder.stream] = transcoder
+
+	for i := 0; i < transcoder.Capacity(); i++ {
+		rtm.remoteTranscoders = append(rtm.remoteTranscoders, transcoder)
+	}
+}
+
+func (rtm *RemoteTranscoderManager) Unregister(t *RemoteTranscoder) {
+	rtm.RTmutex.Lock()
+	delete(rtm.liveTranscoders, t.stream)
+	defer rtm.RTmutex.Unlock()
```

Comment:
removing `defer` here will spare couple of CPU cycles

Reply:
Good catch! I'll remove that